File: packages/http-client-java/generator/http-client-generator-test/src/test/java/com/server/versions/VersionedTests.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

package com.server.versions;

import com.server.versions.versioned.VersionedClient;
import com.server.versions.versioned.VersionedClientBuilder;
import com.server.versions.versioned.VersionedServiceVersion;
import org.junit.jupiter.api.Test;

public class VersionedTests {

    private final VersionedClient client = new VersionedClientBuilder()
            .endpoint("http://localhost:3000")
            .serviceVersion(VersionedServiceVersion.V2022_12_01_PREVIEW)
            .buildClient();

    @Test
    public void test() {
        client.withoutApiVersion();

        client.withPathApiVersion();

        client.withQueryApiVersion();

        VersionedClient oldClient = new VersionedClientBuilder()
                .endpoint("http://localhost:3000")
                .serviceVersion(VersionedServiceVersion.V2021_01_01_PREVIEW)
                .buildClient();
        oldClient.withQueryOldApiVersion();
    }
}

File: packages/http-client-csharp/generator/TestProjects/CadlRanch.Tests/Http/Server/Versions/Versioned/VersionedTests.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System.Threading.Tasks;
using NUnit.Framework;
using Server.Versions.Versioned;

namespace TestProjects.CadlRanch.Tests.Http.Server.Versions.Versioned
{
    public class VersionedTests : CadlRanchTestBase
    {
        [CadlRanchTest]
        public Task WithoutApiVersion() => Test(async (host) =>
        {
            var response = await new VersionedClient(host, null).WithoutApiVersionAsync();
            Assert.AreEqual(200, response.GetRawResponse().Status);
        });

        [CadlRanchTest]
        public Task WithQueryApiVersion() => Test(async (host) =>
        {
            var response = await new VersionedClient(host, null).WithQueryApiVersionAsync();
            Assert.AreEqual(200, response.GetRawResponse().Status);
        });

        [CadlRanchTest]
        public Task WithPathApiVersion() => Test(async (host) =>
        {
            var response = await new VersionedClient(host, null).WithPathApiVersionAsync();
            Assert.AreEqual(200, response.GetRawResponse().Status);
        });

        [CadlRanchTest]
        public Task WithQueryOldApiVersion() => Test(async (host) =>
        {
            var options = new VersionedClientOptions(VersionedClientOptions.ServiceVersion.V2021_01_01_Preview);
            var response = await new VersionedClient(host, options).WithQueryOldApiVersionAsync();
            Assert.AreEqual(200, response.GetRawResponse().Status);
        });
    }
}
